tests/unit/test_option_parser.py

```diff
@@ -191,5 +191,42 @@ def test_link_with_include_paths(self):
     def test_ignore_flags(self):
         ignore = ["-Werror", "-MT hello", "-M", "-fsyntax-only"]
         build_cmd = "g++ {} main.cpp".format(' '.join(ignore))
-        res = option_parser.parse_options(build_cmd)
+        res = option_parser.parse_options(build_cmd, ['main.cpp'])
         self.assertEqual(res.compile_opts, ["-fsyntax-only"])
+
+    def test_keep_warning_flags(self):
+        build_cmd = "g++ -Wall -G 0 -Wextra main.cpp"
+        res = option_parser.parse_options(build_cmd, ['main.cpp'])
+        self.assertEqual(res.compile_opts, ["-Wall", "-G", "0", "-Wextra"])
+
+    def test_unhandled(self):
+        """
+        Unhandled options should be detected as compiler option.
+        """
+        source_files = ["/home/test/main test.cpp"]
+        build_cmd = "g++ -Unhandled " + ' '.join(source_files)
+        res = option_parser.parse_options(build_cmd, source_files)
+        self.assertEqual(res.compile_opts, ["-Unhandled"])
+
+    def test_stuff(self):
```

Comment:
Could you please add more descriptive name for this test? E.g. `test_apostrophe_after_eq_sign`